corehq/form_processor/interfaces/processor.py

```diff
@@ -97,7 +97,8 @@ def ledger_db(self):
     def acquire_lock_for_xform(self, xform_id):
         lock = self.xform_model.get_obj_lock_by_id(xform_id, timeout_seconds=15 * 60)
         try:
-            lock.acquire()
```

Comment:
Previously if this lock could not be acquired (i.e., this line returned `False`) then I think form processing would continue with an unacquired lock. That seems bad, what am I missing? However, it would wait for the lock to expire first, which means it could block for up to 15 minutes.